packages/server-wallet/src/message-service/socket-io-message-service.ts

```diff
@@ -22,13 +30,32 @@ export class SocketIOMessageService implements MessageServiceInterface {
     this.peers.push(socket);
   }
 
+  public setLatencyOptions(options: LatencyOptions): void {
+    this._latencyOptions = options;
+  }
+
   public async send(messages: Message[]): Promise<void> {
-    if (messages.length > 0) {
-      this.server.emit('message', messages);
+    if (this._frozen) {
+      this._messageQueue.push(...messages);
+    } else {
+      const shouldDrop = Math.random() > 1 - this._latencyOptions.dropRate;
+
+      if (!shouldDrop) {
```

Comment:
This is based on the logic in the [TestMessageService](https://github.com/statechannels/statechannels/blob/master/packages/server-wallet/src/message-service/test-message-service.ts)